src/atsamd/adc.c

```diff
@@ -69,7 +95,38 @@ adc_init(void)
         return;
     have_run_init = 1;
 
-#if CONFIG_MACH_SAMD21
+#if CONFIG_MACH_SAMC21
+    // Enable adc clock
+    enable_pclock(ADC0_GCLK_ID, ID_ADC0);
+    enable_pclock(ADC1_GCLK_ID, ID_ADC1);
+
+    // Load calibration info
+    // ADC0
+    uint32_t refbuf = GET_FUSE(ADC0_FUSES_BIASREFBUF);
+    uint32_t comp = GET_FUSE(ADC0_FUSES_BIASCOMP);
+    ADC0->CALIB.reg = (ADC0_FUSES_BIASREFBUF(refbuf)
+                       | ADC0_FUSES_BIASCOMP(comp));
+
+    // ADC1
+    refbuf = GET_FUSE(ADC1_FUSES_BIASREFBUF);
+    comp = GET_FUSE(ADC1_FUSES_BIASCOMP);
+    ADC1->CALIB.reg = (ADC0_FUSES_BIASREFBUF(refbuf)
+                       | ADC0_FUSES_BIASCOMP(comp));
+
+    // Setup and enable
+    // ADC0
+    ADC0->REFCTRL.reg = ADC_REFCTRL_REFSEL_INTVCC1;
+    ADC0->CTRLB.reg = ADC_CTRLB_PRESCALER_DIV128;
+    ADC0->SAMPCTRL.reg = ADC_SAMPCTRL_SAMPLEN(63);
+    ADC0->CTRLA.reg = ADC_CTRLA_ENABLE;
+
+    // ADC1
+    ADC1->REFCTRL.reg = ADC_REFCTRL_REFSEL_INTVCC1;
+    ADC1->CTRLB.reg = ADC_CTRLB_PRESCALER_DIV128;
+    ADC1->SAMPCTRL.reg = ADC_SAMPCTRL_SAMPLEN(63);
+    ADC1->CTRLA.reg = ADC_CTRLA_ENABLE;
+
+#elif CONFIG_MACH_SAMD21
```

Comment:
This code seems very similar to the SAMX5 code - can both the SAMX5 and SAMC21 code share the same code?

Reply:
It is very similar, but I broke it out like this since a section for the sdadc will need to be added.

Reply:
Decided to keep this, it's different enough that the code looks nicer this way, and I may add the sdadc in a future PR.